staging/src/k8s.io/kubectl/pkg/cmd/apply/prune.go

```diff
@@ -70,8 +70,7 @@ func newPruner(o *ApplyOptions) pruner {
 }
 
 func (p *pruner) pruneAll(o *ApplyOptions) error {
-
-	namespacedRESTMappings, nonNamespacedRESTMappings, err := prune.GetRESTMappings(o.Mapper, o.PruneResources, o.Namespace != "")
+	namespacedRESTMappings, nonNamespacedRESTMappings, err := prune.GetRESTMappings(o.Mapper, o.PruneResources, o.EnforceNamespace)
```

Comment:
Doesn't `o.Namespace != ""` already indicate whether a namespace was specified, or is the change to use `o.EnforceNamespace` needed for some reason?

Reply:
`o.Namespace` is `default` if namespace is not specified, so `o.Namespace != ""` is always `true`. `o.EnforceNamespace` is true when the namespace is specified, so I think it's suitable.

https://github.com/kubernetes/kubernetes/blob/b4f902f0371485505ff4eda39975e67bfa9b0727/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#L417-L419

Reply:
`o.Namespace` is `default` if namespace is not specified, so `o.Namespace != ""` is always `true`. `o.EnforceNamespace` is true when the namespace is specified, so I think it's suitable.

https://github.com/kubernetes/kubernetes/blob/b4f902f0371485505ff4eda39975e67bfa9b0727/staging/src/k8s.io/client-go/tools/clientcmd/client_config.go#L417-L419